src/Client/ClientBase.cpp

```diff
@@ -236,15 +236,14 @@ void interruptSignalHandler(int signum)
         _exit(signum);
 }
 
+QueryInterruptHandler query_interrupt_handler;
 
 ClientBase::~ClientBase() = default;
 ClientBase::ClientBase() = default;
 
 
 void ClientBase::setupSignalHandler()
 {
-     exit_on_signal.test_and_set();
```

Comment:
It is not working a little different. Now if you kill the query with sigint just after it started then the program will terminate but it shouldn't. Previously it didn't.

Reply:
Actually this particular case should work, what does not work is second SIGINT since the flag was not reseted, will fix it shortly.

Reply:
> Actually this particular case should work

I start query and then press ctrl-c very very quickly only once right after it stared and the program terminates.

Reply:
Oh, I see, you are right, there is such possibility, good catch!
This should be fixed in the rebased patch.